svm/src/account_loader.rs

```diff
@@ -97,23 +100,39 @@ pub struct FeesOnlyTransaction {
 
 #[cfg_attr(feature = "dev-context-only-utils", derive(Clone))]
 pub(crate) struct AccountLoader<'a, CB: TransactionProcessingCallback> {
-    account_overrides: Option<&'a AccountOverrides>,
     pub(crate) program_cache: ProgramCacheForTxBatch,
     program_accounts: HashMap<Pubkey, (&'a Pubkey, u64)>,
+    account_cache: AHashMap<Pubkey, AccountCacheItem>,
     callbacks: &'a CB,
     pub(crate) feature_set: Arc<FeatureSet>,
 }
 impl<'a, CB: TransactionProcessingCallback> AccountLoader<'a, CB> {
-    pub fn new(
+    pub fn new_with_account_cache_capacity(
         account_overrides: Option<&'a AccountOverrides>,
         program_cache: ProgramCacheForTxBatch,
         program_accounts: HashMap<Pubkey, (&'a Pubkey, u64)>,
         callbacks: &'a CB,
         feature_set: Arc<FeatureSet>,
+        capacity: usize,
     ) -> AccountLoader<'a, CB> {
+        let mut account_cache = AHashMap::with_capacity(capacity);
+
+        if let Some(slot_history) =
+            account_overrides.and_then(|overrides| overrides.get(&slot_history::id()))
+        {
+            // We set `inspected_as_writable` to `true` because this never needs to be inspected.
+            account_cache.insert(
+                slot_history::id(),
+                AccountCacheItem {
+                    account: slot_history.clone(),
+                    inspected_as_writable: true,
```

Comment:
I thought there was a discussion on previous PR about this.
The function that is called to inspect accounts already has guards against doing it twice.

Is this still necessary?

Reply:
we never got confirmation on that. ill check with brooks

Reply:
alright i have clarity on this now, the thing i was missing is that the idea _svm_ has of "what things to inspect" and the idea _bank_ has of "what results of inspection do i actually care about" are disjoint concepts. we can get rid of the state tracking and inspect every time we get an account from accounts-db _or_ account cache. bank has its own dedupe logic and other implementers of inspect can do whatever they want